File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        op = {")":"(", "}":"{", "]":"["}
        stack = []
        for chr in s:
            if chr not in op:
                stack.append(chr)
            else:
                if stack and stack[-1] == op[chr]:
                    stack.pop()
                else:
                    return False
        return len(stack) == 0

File: 20-valid-parentheses/valid-parentheses.go
func isValid(s string) bool {
    stack := []string{}
    pair := map[string]string{
        "(": ")",
        "{": "}",
        "[": "]",
    }

    for _, par := range s {
        char := string(par)
        if closing, exists := pair[char]; exists {
            stack = append(stack, closing)
        } else {
            if len(stack) == 0 || stack[len(stack)-1] != char {
                return false
            }
            stack = stack[:len(stack)-1]
        }
    }
    return len(stack) == 0
}